### PR TITLE
feat(useFetch): payload supports ref and follow changed

### PR DESCRIPTION
**Pull request by Jay214**
Set automatically refetch if the payload is a ref and changed.
This is a common scenario where we want to make the request again when the parameters change.

### CODE REVIEW

File: packages/core/useFetch/index.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export interface UseFetchOptions {
   immediate?: boolean
 
   /**
-   * Will automatically refetch when the URL is changed if the url is a ref
+   * Will automatically refetch when the URL is changed if the url is a ref or the payload is a ref
```
Comment (DarrylNoakes) [inline packages/core/useFetch/index.ts:132]:
```suggestion
   * Will automatically refetch when:
   * - the URL is changed if the URL is a ref
   * - the payload is changed if the payload is a ref
```

File: packages/core/useFetch/index.ts [hunk 2]
```diff
@@ -407,9 +412,10 @@ export function useFetch<T>(url: MaybeRef<string>, ...args: any[]): UseFetchRetu
         config.method = method
         config.payload = payload
         config.payloadType = payloadType
+        watchPayload()
         // Set the payload to json type only if it's not provided and a literal object is provided
         // The only case we can deduce the content type and `fetch` can't
-        if (!payloadType && payload && Object.getPrototypeOf(payload) === Object.prototype)
+        if (!isRef(payload) && !payloadType && payload && Object.getPrototypeOf(payload) === Object.prototype)
```
Comment (DarrylNoakes) [inline packages/core/useFetch/index.ts:418]: What does this do?

Why not:

```suggestion
        if (!payloadType && unref(payload) && Object.getPrototypeOf(unref(payload)) === Object.prototype)
```

File: packages/core/useFetch/index.md [hunk 3]
```diff
@@ -235,7 +235,8 @@ export interface UseFetchOptions {
    */
   immediate?: boolean
   /**
-   * Will automatically refetch when the URL is changed if the url is a ref
+   * Will automatically refetch when the URL is changed if the url is a ref 
+   * or the payload is a ref and changed
```
Comment (DarrylNoakes) [inline packages/core/useFetch/index.md:239]:
```suggestion
   * Will automatically refetch when:
   * - the URL is changed if the URL is a ref 
   * - the payload is changed if the payload is a ref
```

**Review by DarrylNoakes — CHANGES_REQUESTED** [review]
Just some suggestions.
Take them as you please 😉

**Comment by Jay214** [comment]
> Just some suggestions.
> Take them as you please 😉

thanks you suggestions. is useful

File: packages/core/useFetch/index.ts (hunk 2 above)
Comment (Jay214) [reply packages/core/useFetch/index.ts:418]: In fact, 'Object.getPrototypeOf(payload) === Object.prototype' will always return false if the payload is a ref

**Pull request merged**